### DISCUSSION TITLE
New features and FSRS-6

### DISCUSSION THREAD

**Discussion by Expertium** [open-spaced-repetition/free-spaced-repetition-scheduler#30] [Ideas]
@L-M-Sherlock 
https://forums.ankiweb.net/t/revisiting-onboarding-decks-and-tutorials/43692/96?u=expertium
> some developers have complained that FSRS changes its formula too frequently. FSRS has a dozen of packages. Once we update the formula, they all need to keep up the latest version, and it has blocked their roadmap/plan/todos several times. So I just want to slow down. And the name would not be FSRS-6 because we may have a series of variants of FSRS to satisfy different constrains.

So what would you call the new version? IMO it's fine to call it FSRS-6. I don't plan to make any further changes after [this](https://github.com/open-spaced-repetition/srs-benchmark/issues/125), and your are "retiring" (or going on an indefinite hiatus, I suppose), so there won't be frequent changes anymore.

**Comment by L-M-Sherlock** [comment]
For example, we may have a version that support workload and hours as the input features only, and another version that support precise short-term memory model only. It's not a good idea to name them FSRS-6 and FSRS-7.

**Reply by Expertium** [reply]
Ok, but what would you call them? I can't think of a name that is concise and conveys the difference. FSRS-5-shortterm is too long, FSRS-5-fourinputs is awful.
@user1823 any good suggestions?

**Reply by NamelessGO** [reply]
> support workload and hours
- FSRSx-L: Long (hours, days) Focused
- FSRSx: Without affix, pairs nicely all below

> precise short-term memory model only
- FSRSx-S: Short focused
- FSRSx-P: Precise
-  FSRSx-Lite


x = Version Number

-----
Only noticed after that the Version number goes after dash, FSRS-5 🤡 :, but these are some name recommendations

**Reply by Expertium** [reply]
So like FSRS-5-L? Ok, not bad. I would rename it to FSRS-5-F where the last "F" stands for "Fatigue", which is new variable that FSRS will be using. And then FSRS-5 with a better formula for same-day reviews would be FSRS-5-S.
Jarrett, what do you think?
FSRS-5 for the current version; FSRS-5-S for the version that you are working on, with better formulas for same-day reviews; FSRS-5-F for the one I will be working on, with Fatigue.

**Reply by brishtibheja** [reply]
I thought there is something intersting to give opinions on, but lol. Anyways, go with these names if you expect the different versions to have a consistent "base algorithm". Otherwise, I think we shouldn't have "5" in the naming. The next version of FSRS's fatigue version than will be weirdly called for FSRS-5-F2. My suggestion is to move back to version one (like GPT-o did).

If everyone hates that idea, then FSRSf-5 is also possible. This allows names like FSRSf-5.1 for minor versions and change in the major number (5 here) when the base algorithm is changing.
